### PR TITLE
release: v2.63.0-rc.7 — restore connected-device RSSI via mgmt 0x0031

### PR DESCRIPTION
**Pull request by trudenboy**
## Summary

Restores the live RSSI display for connected speakers — third (and working) attempt after rc.3 (`scan bredr`) and rc.5 (`bluetoothctl info`) both proved BlueZ exposes nothing for an established ACL link via userspace tools.

The remaining viable Linux source is the kernel mgmt socket: opcode 0x0031 (`MGMT_OP_GET_CONN_INFO`). Wrapped in `services/bt_rssi_mgmt.py` via the `btsocket` library so we don't hand-roll the binary protocol. Requires `CAP_NET_ADMIN` — bridge container already has it.

- `services/bt_rssi_mgmt.read_conn_info` — sync wrapper. Every failure mode (peer not connected, EPERM, ENODEV, status != Success, sentinel 127, btsocket missing on non-Linux test envs, garbage adapter index) collapses to `None` so callers' contract is "fresh value or keep last known, never raise".
- `BluetoothManager.run_rssi_refresh_loop` + `_rssi_refresh_tick` — every 30 s, gated by the shared `bt_operation_lock` so a pair / scan / reconnect can never starve. Spawned alongside `monitor_and_reconnect` in `sendspin_client._run_async`.
- `on_rssi_update` callback wired through `services/device_activation.activate_device` to `SendspinClient._update_status({"rssi_dbm": ..., "rssi_at_ts": ...})`. Values flow into the existing SSE pipeline and the rc.2 `_renderRssiChip` UI chip without further frontend changes.
- `btsocket==0.3.0` added to `requirements.txt`. `ImportError`-tolerant so macOS dev test runs without it stay green.

## Test plan

- [x] `pytest tests/test_bt_rssi_mgmt.py` — 8 passing (every short-circuit + unsigned->signed byte fold + sentinel 127)
- [x] `pytest tests/test_bt_rssi_refresh.py` — 9 passing (adapter-index resolution + every refresh-tick branch incl. lock contention, callback exceptions, lock release on raise)
- [x] `pytest -q` — full suite 1680 passing
- [x] `ruff check` clean
- [ ] Manual on VM 105: confirm RSSI chip populates within 30 s of connect, updates each tick, greys out at 90 s if mgmt starts failing
- [ ] Manual on HAOS 104: same, plus verify no contention with concurrent pair attempt

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: services/bt_rssi_mgmt.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    raw = getattr(getattr(rsp, "cmd_response_frame", None), "rssi", None)
+    if raw is None:
+        return None
+
+    # btsocket reports the wire byte unsigned; mgmt-api.txt declares it signed.
+    if raw > 127:
+        raw -= 256
+    if raw == _RSSI_UNAVAILABLE_SENTINEL or raw == -_RSSI_UNAVAILABLE_SENTINEL - 1:
+        return None
```
[the remark this answers is not in the document]
Comment (trudenboy) [reply services/bt_rssi_mgmt.py]: Good catch — fixed in b4f9e324. Sentinel check moved before the unsigned→signed fold so raw 0x80 → −128 dBm is preserved as a real (very weak) reading, only raw 127 is treated as "unavailable". New test `test_does_not_drop_minus_128_as_unavailable` pins it.

File: bluetooth_manager.py [hunk 2]
```diff
@@ -1186,6 +1200,76 @@ def _publish_client_event(
...
+            return
+        if not _bt_op_lock.try_acquire_bt_operation():
+            return
+        try:
+            adapter_index = self._resolve_adapter_index()
+            mac = self.mac_address
+            loop = asyncio.get_running_loop()
+            try:
+                rssi = await loop.run_in_executor(_bt_executor, bt_rssi_mgmt.read_conn_info, adapter_index, mac)
+            except Exception:
```
[the remark this answers is not in the document]
Comment (trudenboy) [reply bluetooth_manager.py:1251]: Agreed — fixed in b4f9e324. Both `on_rssi_update is None` and `adapter_index < 0` short-circuits moved before the `try_acquire_bt_operation` call so the every-30-s tick is a true no-op for those cases. Two new tests verify the lock stays free in both paths.
